File: src/main/java/com/sfh/pokeRogueBot/model/exception/UnsupportedPhaseException.java
package com.sfh.pokeRogueBot.model.exception;

import com.sfh.pokeRogueBot.model.enums.UiMode;

public class UnsupportedPhaseException extends RuntimeException {

    public UnsupportedPhaseException(String phase, UiMode gameMode) {
        super("Phase " + phase + " with GameMode " + gameMode + " is not supported yet");
    }

    public UnsupportedPhaseException(String phase) {
        super("Phase " + phase + " is not supported yet");
    }
}

File: src/main/kotlin/com/sfh/pokeRogueBot/model/exception/UnsupportedPhaseException.kt
package com.sfh.pokeRogueBot.model.exception

import com.sfh.pokeRogueBot.model.enums.UiMode

class UnsupportedPhaseException : RuntimeException {
    constructor(
        phase: String,
        gameMode: UiMode
    ) : super("Phase " + phase + " with GameMode " + gameMode + " is not supported yet")

    constructor(phase: String) : super("Phase " + phase + " is not supported yet")
}